org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/rendering/ResourceElementTests.java

```diff
@@ -0,0 +1,226 @@
+    Assertions.assertEquals("identifier", id.name());
+    Assertions.assertEquals("Bundle.identifier", id.path());
+    Assertions.assertFalse(id.isPrimitive());
+    Assertions.assertFalse(id.hasPrimitiveValue());
+    Assertions.assertTrue(id.hasChildren());
+    Assertions.assertEquals(ElementKind.DataType, id.kind());
+
+    ResourceElement system = id.child("system");
+    Assertions.assertEquals("uri", system.fhirType());
+    Assertions.assertEquals("system", system.name());
+    Assertions.assertEquals("Bundle.identifier.system", system.path());
+    Assertions.assertTrue(system.isPrimitive());
+    Assertions.assertTrue(system.hasPrimitiveValue());
+    Assertions.assertEquals("http://something1", system.primitiveValue());
+    Assertions.assertFalse(system.hasChildren());
+    Assertions.assertEquals(ElementKind.PrimitiveType, system.kind());
+
+    ResourceElement value = id.child("value");
+    Assertions.assertEquals("string", value.fhirType());
+    Assertions.assertEquals("value", value.name());
+    Assertions.assertEquals("Bundle.identifier.value", value.path());
+    Assertions.assertTrue(value.isPrimitive());
+    Assertions.assertTrue(value.hasPrimitiveValue());
+    Assertions.assertEquals("something2", value.primitiveValue());
+    Assertions.assertFalse(value.hasChildren());
+    Assertions.assertEquals(ElementKind.PrimitiveType, value.kind());
+
+    int i = 0;
+    for (ResourceElement link : bnd.children("link")) {
+      checkLink(i, link);
+      i++;
+    }
+
+    ResourceElement entry = bnd.child("entry");
+    Assertions.assertEquals("Bundle.entry", entry.fhirType());
+    Assertions.assertEquals("entry", entry.name());
+    Assertions.assertEquals("Bundle.entry[0]", entry.path());
+    Assertions.assertFalse(entry.isPrimitive());
+    Assertions.assertFalse(entry.hasPrimitiveValue());
+    Assertions.assertTrue(entry.hasChildren());
+    Assertions.assertEquals(ElementKind.BackboneElement, entry.kind());
+
+    ResourceElement fu = entry.child("fullUrl");
+    Assertions.assertEquals("uri", fu.fhirType());
+    Assertions.assertEquals("fullUrl", fu.name());
+    Assertions.assertEquals("Bundle.entry[0].fullUrl", fu.path());
+    Assertions.assertTrue(fu.isPrimitive());
+    Assertions.assertTrue(fu.hasPrimitiveValue());
+    Assertions.assertEquals("http://something5", fu.primitiveValue());
+    Assertions.assertFalse(fu.hasChildren());
+    Assertions.assertEquals(ElementKind.PrimitiveType, fu.kind());
+
+    ResourceElement obs = entry.child("resource");
+    checkObservation(obs);
+  }
+
+  private void checkObservation(ResourceElement obs) {
+    Assertions.assertTrue(obs.fhirType().equals("Observation"));
+    Assertions.assertEquals("resource", obs.name());
+    Assertions.assertEquals("Bundle.entry[0].resource", obs.path());
+    Assertions.assertEquals(ElementKind.BundleEntry, obs.kind());
+    
+    List<ResourceElement> children = obs.children();
+    assertEquals(3, children.size());
```

Comment:
Something doesn't seem quote right here. Are `id` and `text` not children?

Reply:
I added those properties to the test case - will update the source for the test